dot/node.go

```diff
@@ -170,7 +172,7 @@ func NodeInitialized(datadir string, expected bool) bool {
 }
 
 // NewNode creates a new dot node from a dot node configuration
-func NewNode(cfg *Config, ks *keystore.Keystore) (*Node, error) {
+func NewNode(cfg *Config, ks *keystore.Keystore, ctx *cli.Context) (*Node, error) {
```

Comment:
`name` and `id` can be found in the `Config`: https://github.com/ChainSafe/gossamer/blob/development/dot/config.go#L45
I think it would make more sense to put the version into the config, and then start the RPC service passing that info, rather than passing the `*cli.Context`, since the `*cli.Context` should only be used when the node is starting up

Reply:
I was going to put system_name, system_version and system_chain (chain type) into the config, but I didn't because it seems that things in config can be over written by the .toml file, and these values should not be able to be set by config.  I agree I'd rather not keep a reference to `*cli.Context`, so I need to store to store App.name and App.version somewhere for system_name and system_version.  Should I store this in config? or add to one of the services?  Which one?  Regarding name in config, I'm using that for system_chain although I not sure that is correct.  Where can I find chain related values (system_chain, and system_properties: ss58Format, tokenDecimals, and tokenSymbol)?